core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -258,6 +258,30 @@ Kokkos::HIP::size_type *HIPInternal::scratch_functor(
   return m_scratchFunctor;
 }
 
+Kokkos::HIP::size_type *HIPInternal::scratch_functor_host(
+    const std::size_t size) const {
+  if (verify_is_initialized("scratch_functor_host") &&
+      m_scratchFunctorSize < size) {
+    m_scratchFunctorSize = size;
```

Comment:
args this is actually wrong. If you first call `scratch_functor()` and then `scratch_functor_host()` the first call will reset m_scratchFunctorSize so the second will not do anything ...